src/component/FolderToolBar/FolderToolBar.jsx

```diff
@@ -1,47 +1,94 @@
-import React, { useState, useEffect } from "react";
+import { useState } from "react";
 import FolderButton from "./FolderButton/FolderButton";
 import FolderCardList from "../FolderToolBar/FolderCardList/FolderCardList";
+import ModalEdit from "../Modal/ModalEdit";
+import ModalShare from "../Modal/ModalShare";
+import ModalDelete from "../Modal/ModalDelete";
+import { ALL } from "../../utils/utils";
 import "./FolderToolBar.css";
+import share from "../../images/share.svg";
+import pen from "../../images/pen.svg";
+import deleteIcon from "../../images/delete.svg";
 
-const FolderToolBar = () => {
-  const [buttons, setButtons] = useState([]);
-  const [links, setLinks] = useState([]);
+const FolderToolBar = ({
+  folderNameData,
+  links,
+  selectedButtonName,
+  onFolderSelect,
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalType, setModalType] = useState(null);
```

Comment:
그런데 이 로직이 모달을 사용하는 컴포넌트마다 반복되네요 ㅎㅎ 훅으로 분리해보면 어떨까요?
아래는 그냥 예시이니 참고해봐주세요 ㅎㅎ
```suggestion
  const useModal = () => {
       const [isModalOpen, setIsModalOpen] = useState(false);
      const [modalType, setModalType] = useState(null);
      
      const openModal =() => {
           setIsModalOpen(true);
       }
       
       const closeModal =() => {
           setIsModalOpen(false);
       }
       
       const changeModalType =(type)=>{
           setModalType(type);
       }
       
       return {isModalOpen, openModal, closeModal, modalType, changeModalType};
  
   }
```